Show mksquashfs progress without being too verbose

afbjorklund opened the pull request:
If the mksquashfs version is new enough (version 4.6 or later),
then show the percentage of progress only if not already verbose.

If the mksquashfs version is older, than fallback to the old message:
"To see mksquashfs output with progress bar enable verbose logging"

## Description of the Pull Request (PR):

Show a progress bar for the SIF creation, including percentage and ETA.

This is just one line of output, compared to the `--verbose` multiple lines.

### This fixes or addresses the following GitHub issues:

* Fixes #2787

#### Before submitting a PR, make sure you have done the following:

- [x] Read the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [x] Make sure all commits are signed-off with `git commit -s`, see the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [x] Added changes to the [CHANGELOG](https://github.com/apptainer/apptainer/blob/main/CHANGELOG.md), if necessary according to the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [ ] Added tests to validate this PR, linted with `make check`  and tested this PR locally with a `make test`, and `make testall` if possible (see the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md)).
- [x] Based this PR against the appropriate branch according to the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [x] Added myself as a contributor to the [Contributors File](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTORS.md).

DrDaveD reviewed (changes requested):
Very cool, thanks.  Please add a CHANGELOG entry in the post 1.4.0-rc.1 section.  When 1.4.0 is fully released I'll combine it with the entry describing progress in verbose mode.

DrDaveD commented:
@afbjorklund Are you happy with this now or do you still consider it a draft?

afbjorklund commented:
> @afbjorklund Are you happy with this now or do you still consider it a draft?

It should be good to go, I added the Done for completion and updated CHANGELOG

DrDaveD commented:
Please also submit a cherry-pick PR to release-1.4
